### PR TITLE
Docs Design

### PR DESCRIPTION
**Pull request by marvinhagemeister**
Still lots of stuff to do, but opening this for others to look at while it progresses.

- [x] Fix sidebar active page animation not triggering (the dot should move)
- [x] Add dark mode switch
- [ ] Add search
- [ ] Add responsive layout
- [ ] Add home page
- [ ] Swap out "home" in header with a logo
- [ ] Add aria current page markers
- [ ] Update code syntax colors to match theme
- [x] Add back blockquote styles
- [ ] ...

### CODE REVIEW

File: docs/public/lib/use-localstorage.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { useState } from 'preact/hooks';
+
+/**
+ * @type {<T>(name: string, value: T) => [T, (v: T) => void]}
+ */
+export function useLocalStorage(key, initial) {
+	const [v, setValue] = useState(() => {
+		const stored = localStorage.getItem(key);
```
Comment (JoviDeCroock) [inline docs/public/lib/use-localstorage.js:8]: Hmm, does this work with pre-rendering? The error in netlify refers to this being undefined

File: docs/public/styles/index.css [hunk 2]
```diff
@@ -74,6 +74,9 @@ html[theme='dark'] {
 
 	--header-accent-height: 0.1875rem;
 }
+html[theme='dark'] .icon {
+	filter: invert(1);
```
Comment (marvinhagemeister) [inline docs/public/styles/index.css:78]: We need to add this to the "prefers-color-scheme: dark" block too

Comment (developit) [reply docs/public/styles/index.css:78]: It's there:

https://github.com/preactjs/wmr/pull/503/files#diff-c55bbfb39ca79e1c551b60cf79f7dcb1c40fba2788af584db3e739913cbe9f38R47

Comment (marvinhagemeister) [reply docs/public/styles/index.css:78]: whoops, my bad 🙈

**Comment by marvinhagemeister** [comment]
As discussed on slack we'll merge this, so that other PRs can begin to add more doc content while we iterate on the general layout and design.

**Pull request merged**